.github/workflows/workflow.yml

```
@@ -0,0 +1,75 @@
+name: Main workflow
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
+          - windows-latest
+        # Please keep the list in sync with the minimal version of OCaml in
+        # dune-project, dune.opam.template and bootstrap.ml
+        ocaml-version:
+          - 4.10.0
+        skip_test:
+          - false
+        include:
+          - ocaml-version: 4.09.1
+            os: ubuntu-latest
+            skip_test: true
+          - ocaml-version: 4.08.1
+            os: ubuntu-latest
+            skip_test: true
+          - ocaml-version: 4.03.0
+            os: ubuntu-latest
+            skip_test: true
+          - ocaml-version: 4.02.3
+            os: ubuntu-latest
+            skip_test: true
+
+    # We don't run tests on all versions of the Windows environment and on
+    # 4.02.3 and 4.07.1 in other environments
+    env:
+      # Note that this environment variable is recognized as a string by GitHub
+      # Actions, so the following doesn't actually work:
+      #
+      # if: env.SKIP_TEST
+      #
+      SKIP_TEST: ${{ matrix.skip_test }}
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Use OCaml ${{ matrix.ocaml-version }}
+        uses: avsm/setup-ocaml@v1
+        with:
+          ocaml-version: ${{ matrix.ocaml-version }}
+
+      # git user needs to be configured for the following tests:
+      # otherlibs/build-info/test/run.t
+      # test/blackbox-tests/test-cases/dune-project-meta/run.t
+      # test/blackbox-tests/test-cases/subst/run.t
+      # test/expect-tests/vcs_tests.ml
+      - name: Set git user
+        run: |
+          git config --global user.name github-actions
+          git config --global user.email github-actions-bot@users.noreply.github.com
+
+      - run: opam exec -- ocaml .github/workflows/ci.ml pin
+
+      - run: opam install ./dune.opam --deps-only --with-test
```

```suggestion
      # Install ocamlfind-secondary and ocaml-secondary-compiler, if needed
      - run: opam install ./dune.opam --deps-only --with-test
```
(at a glance, the presence of "install the deps of dune" conflicted with the comment below that Dune doesn't have any dependencies!)